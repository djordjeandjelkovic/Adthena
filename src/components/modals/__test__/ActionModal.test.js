import React from 'react';
import ActionModal from '../ActionModal';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe("<ActionModal /> test cases", () => {
  const open = true;
  const setOpen = jest.fn();
  const title = 'Action modal title';
  const confirmAction = jest.fn();

  test('test with all passed elements', () => {
    const deleteTitle = 'Delete user';
    const confirmButtonText = 'Delete';
    const cancelButtonText = 'Cancel';
    const children = <p>Are you sure you want to delete user?</p>;

    const { getByTestId, container } = render(<ActionModal
        open={open}
        setOpen={setOpen}
        title={deleteTitle}
        confirmAction={confirmAction}
        confirmButtonText={confirmButtonText}
        cancelButtonText={cancelButtonText}>
        {children}
    </ActionModal>);
    
    expect(getByTestId('confirm-button')).toHaveTextContent(confirmButtonText);
    expect(getByTestId('cancel-button')).toHaveTextContent(cancelButtonText);
    expect(getByTestId('title')).toHaveTextContent(deleteTitle);
    expect(container).toContainHTML("<p>Are you sure you want to delete user?</p>");
  });

  test('test default values', () => {
    const title = 'Delete user';    
    const defaultConfirmButtonText = 'Confirm';
    const defaultCancelButtonText = 'Cancel';

    const { getByTestId } = render(<ActionModal
        open={open}
        setOpen={setOpen}
        title={title}
        confirmAction={confirmAction} />);

    expect(getByTestId('confirm-button')).toHaveTextContent(defaultConfirmButtonText);
    expect(getByTestId('cancel-button')).toHaveTextContent(defaultCancelButtonText);
  });

  test('matches snapshot', () => {
    const title = 'Action modal title';
    const confirmButtonText = 'Confirm';
    const cancelButtonText = 'Cancel';

    const tree = renderer.create(<ActionModal
        open={open}
        setOpen={setOpen}
        title={title}
        confirmAction={confirmAction}
        confirmButtonText={confirmButtonText}
        cancelButtonText={cancelButtonText} />
    );

    expect(tree).toMatchSnapshot();
  });

  test('confirm button click', () => {    
    const { getByTestId } = render(<ActionModal
        open={open}
        setOpen={setOpen}
        title={title}
        confirmAction={confirmAction} />);
    getByTestId('confirm-button').click();
    expect(confirmAction).toBeCalledTimes(1);
  });

  test('cancel button click', () => {    
    let modal = render(<ActionModal
        open={open}
        setOpen={setOpen}
        title={title}
        confirmAction={confirmAction} />);

    const { getByTestId } = modal;
    getByTestId('cancel-button').click();
    expect(setOpen).toBeCalledTimes(1);
  });
});
