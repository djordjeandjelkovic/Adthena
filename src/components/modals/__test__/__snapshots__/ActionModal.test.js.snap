// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionModal /> test cases matches snapshot 1`] = `
<div
  className="Task__ModalBackground-sc-7z955n-4 bcpqJ"
  onClick={[Function]}
>
  <div
    className="Task__ModalWrapper-sc-7z955n-5 bVHENH"
  >
    <div
      className="Task__ModalTitle-sc-7z955n-6 iXNurm"
      data-testid="title"
    >
      <img
        alt="alert"
        className="Task__ModalTitleAlert-sc-7z955n-7 dTZvCu"
        src="images/alert.jpg"
        width="20"
      />
      <strong>
        Action modal title
      </strong>
    </div>
    <div
      className="Task__ModalContent-sc-7z955n-8 jtmxWF"
    />
    <div
      className="Task__ModalAction-sc-7z955n-9 byNCQS"
    >
      <button
        className="Task__ButtonConfirm-sc-7z955n-10 igynex"
        data-testid="confirm-button"
        onClick={[MockFunction]}
      >
        <strong>
          Confirm
        </strong>
      </button>
      <button
        className="Task__ButtonCancel-sc-7z955n-11 zWqzm"
        data-testid="cancel-button"
        onClick={[Function]}
      >
        <strong>
          Cancel
        </strong>
      </button>
    </div>
  </div>
</div>
`;
